pkg/ccl/sqlproxyccl/forwarder.go

```diff
@@ -346,15 +344,26 @@ func (p *processor) resume(ctx context.Context) error {
 		p.mu.inPeek = false
 		return p.mu.suspendReq
 	}
+	updateLastMessage := func(typ byte) (terminate bool) {
```

Comment:
I think we should merge updateLastMessage with exitPeek. See https://gist.github.com/JeffSwenson/80cd66bf805bf20670867d26e99f44f1 as an example. Reducing the number of critical sections reduces the probability of bugs like this slipping in.